build-tools/src/main/java/org/elasticsearch/gradle/testclusters/RunTask.java

```diff
@@ -165,6 +165,7 @@ public void runAndWait() throws IOException {
         if (initOnly) {
             return;
         }
+        getClusters().forEach(ElasticsearchCluster::waitForAllConditions);
```

Comment:
this is the actual fix to support multi-node via run. This causes the ports files to be written out to disk that is needed to form the cluster.

Reply:
This might be problematic if something explodes in startup, because this call blocks, we won't get any log output until the cluster reports green, which might be never. Should we move this down below?

Reply:
good catch.  All I really need is that call to update the ports information. I removed this blocking line and just called the method to write out the ports info. [`23b4be1` (#89563)](https://github.com/elastic/elasticsearch/pull/89563/commits/23b4be1305be16f7c5b4f1db7b24e2b47f0ad4a0)